### PR TITLE
add validation methods

### PR DESCRIPTION
**Pull request by bj00rn**
Add methods for validation to support future implementation validation of parameters and api key in Home Assistant `config flow` in order to avoid creating broken sensors. To partially support https://github.com/home-assistant/core/issues/106771

* validate api key (validation success not tested since I do not have an api key atm)
  -> (https://doc.forecast.solar/for_developers)
* validate plane -> (https://doc.forecast.solar/api:misc)

I had to add `rate_limit` and `authenticate` flags to `_request` method since `/check` endpoint does not provide rate limit headers in response and does not seem to exist behind authentication (at least not according to the docs)

### CODE REVIEW

**Comment by bj00rn** [comment]
Seems that `{api_key}/info` endpoint does not provide response `code` when `401` is returned for an invalid key, this causes an exception to be thrown in https://github.com/home-assistant-libs/forecast_solar/blob/171deff67b95e80605806e3db4800e4a01ceddb6/forecast_solar/exceptions.py#L29 when key is invalid. I've just tested with a nonsense key `aaaaaaaaaaaaaaaa` though som maybe results may vary dependending if the key is well formatted or not.

Maybe it's worth into making `code` optional for exceptions

**Comment by bj00rn** [comment]
Depends on Please merge https://github.com/home-assistant-libs/forecast_solar/pull/43 for linting errors

**Comment by bj00rn** [comment]
@frenck @MartinHjelmare rebased now, tests are passing

File: forecast_solar/__init__.py [hunk 1]
```diff
@@ -144,6 +150,33 @@ async def _request(
 
         return await response.json()
 
+    async def validate_plane(self):
```
Comment (klaasnicolaas) [inline forecast_solar/__init__.py:153]:
```suggestion
    async def validate_plane(self) -> bool:
```

File: forecast_solar/__init__.py [hunk 2]
```diff
@@ -144,6 +150,33 @@ async def _request(
...
+        await self._request(
+            f"check/{self.latitude}/{self.longitude}"
+            f"/{self.declination}/{self.azimuth}/{self.kwp}",
+            False,
+            False,
+        )
+
+        return True
+
+    async def validate_api_key(self):
```
Comment (klaasnicolaas) [inline forecast_solar/__init__.py:169]:
```suggestion
    async def validate_api_key(self) -> bool:
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: forecast_solar/__init__.py (hunk 1 above)
Comment (bj00rn) [reply forecast_solar/__init__.py:153]: fixed

File: forecast_solar/__init__.py (hunk 2 above)
Comment (bj00rn) [reply forecast_solar/__init__.py:169]: fixed

**Comment by bj00rn** [comment]
@klaasnicolaas any chance of getting this merged soon? I Need this to resolve https://github.com/home-assistant/core/issues/106771

File: forecast_solar/__init__.py [hunk 3]
```diff
@@ -144,6 +150,33 @@ async def _request(
...
+        return True
+
+    async def validate_api_key(self) -> bool:
+        """Validate api key by calling the Forecast.Solar API
+
+        Returns:
+            True, if api key is valid
+        """
+
+        await self._request("info", False)
```
[the remark this answers is not in the document]
Comment (bj00rn) [reply forecast_solar/__init__.py]: fixed

**Review by klaasnicolaas — APPROVED** [review]

**Comment by bj00rn** [comment]
@klaasnicolaas Thanks! Will you carve a new release with this fix?

**Comment by klaasnicolaas** [comment]
@bj00rn Yes, release v3.1.0 is now available.

**Pull request merged**